### DISCUSSION TITLE
Documentation is slightly incorrect

### DISCUSSION THREAD

**Discussion by duckdoom4** [DRincs-Productions/pixi-vn#274] [Wiki chat (giscus)]
Hi, out of curiosity I was taking a look at the [Ren'Py vs Pixi'VN](https://pixi-vn.web.app/start/why.html#programming-language) section in the documentation.

I noticed the following statements:

> Programming language
> 
> Ren'Py uses its own language, Ren'Py language, which is based on Python. This language is:
> 
>     Very simple and easy to learn, suitable for people who are not programmers, but it is not suitable for creating complex games.
>     Not typed language
>     At the moment, there are no debug systems present.
>     Only used in RenPy, so once you learn it you will have no way to use it for other types of development.

But part of this is incorrect. 
* It's not just based on python, it's actually a superset of python. Meaning 'advanced' programmers can actually code their entire game in pure python.
* It supports python 3, which optionally supports type annotations (not just in comments).
* That said, you basically learn python if you learn the renpy language

Just thought I'd point this out, cus that makes the comparison a bit unfair due to misinformation.

**Comment by BlackRam-oss** [comment]
Hi sorry for the delay, but I didn't notice this issue.

Ok, it is "It's not just based on python, it's actually a superset of python".

But for the rest I don't really agree with you.
For a while I used renpy creating and using libraries in python. This often led me to read the renpy code. And to add "renpy native" as if it were a library in the python libraries folder (in this way the python VSCode extension suggested me the renpy functions in python).

What I noticed are the following things:
* I could use types when using python files. and I think that in Python Statements you can use types with the help of the VSCode extension (I don't know if this possibility is already available). Yes I should say this in the wiki.
* Inside renpy language (not in python) you cannot assign types to variables
* I really rarely saw functions that exploited types in the native renpy code.
* The renpy compiler does not do any type checking. (It is possible that a type check will be added in the future through a VScode extension, but I do not think it will be added by the renpy compiler)
* The times I asked questions on the renpy discord they always advised me not to use python but to write the code in renpy

I can add in the wiki that if you use python you can use types. but I should also say that using python files is not recommended and that there is no compile-time checking in Python Statements.

Anyway it makes sense to discuss it, I could be wrong.

That page will have to be revised because new features have been introduced on the pixi'vn side so this is the best time. Thanks to a recent edit, now, you can potentially integrate any narrative language (even if it's not that simple) for now ink has been added

**Comment by BlackRam-oss** [comment]
@duckdoom4 Could you tell me if you have any other observations on the comparison with Renpy?

**Comment by duckdoom4** [comment]
I think the points you made are valid. It wasn't my intention to be the 'well actually' guy just to point out you might have missed something. 
But you clearly have used renpy and know what you're talking about, so I trust you can make a fair comparison. Though I would say: Add that info you just told me to the comparison paragraph, in my experience this type of information is very important when deciding what to use

**Comment by BlackRam-oss** [comment]
Ok, anyway I created a multiplication table that I think makes more sense. (I tried to be unbiased) https://github.com/DRincs-Productions/pixi-vn/wiki/why#programming-language
